Makefile

```diff
@@ -1,321 +1,52 @@
-
-${TMPDIR}:
-	mkdir -p ${TMPDIR}
-
-maybe-prepare:
-ifeq ($(SKIP_KIND_SETUP), 0)
-	$(MAKE) prepare
-endif
-
-clean:
-	docker run --privileged \
-		-v /var/run/docker.sock:/var/run/docker.sock  \
-		-e GOPATH=${GOPATH} \
-		-it --entrypoint /bin/bash --rm \
-		$(BUILD_IMAGE) -c \
-		"/usr/local/bin/kind delete cluster --name ${KIND_CLUSTER}" 2>&1 || true
-	#kind delete cluster --name ${KIND_CLUSTER} 2>&1 || true
-
-maybe-clean:
-ifeq ($(SKIP_CLEANUP), 0)
-	$(MAKE) clean
-endif
-
-# Generate junit reports, upload to testgrid, fail if conditions are met.
-# Failure is based on test status - may exclude some tests.
-report:
-	bin/report.sh
-
-info:
-	# Get info about env, for debugging
-	set -x
-	pwd
-	env
-	id
-
-
-# Copy source code from the current machine to the docker.
-sync:
-ifneq ($(MOUNT), 1)
-	docker exec ${KIND_CLUSTER}-control-plane mkdir -p ${GOPATH}/src/istio.io/installer \
-		${GOPATH}/src/istio.io
-	docker cp . ${KIND_CLUSTER}-control-plane:${GOPATH}/src/istio.io/installer
+# WARNING: DO NOT EDIT, THIS FILE IS PROBABLY A COPY
+#
+# The original version of this file is located in the https://github.com/istio/common-files repo.
+# If you're looking at this file in a different repo and want to make a change, please go to the
+# common-files repo, make the change there and check it in. Then come back to this repo and run
+# "make updatecommon".
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+RUN =
+
+# Set the enviornment variable BUILD_WITH_CONTAINER to use a container
+# to build the repo. The only dependencies in this mode are to have make and
+# docker. If you'd rather build with a local tool chain instead, you'll need to
+# figure out all the tools you need in your environment to make that work.
+export BUILD_WITH_CONTAINER ?= 0
+ifeq ($(BUILD_WITH_CONTAINER),1)
+IMG = gcr.io/istio-testing/build-tools:2019-08-05
+UID = $(shell id -u)
+PWD = $(shell pwd)
+GOBIN_SOURCE ?= $(GOPATH)/bin
+GOBIN ?= /work/out/bin
+
+RUN = docker run -t --sig-proxy=true -u $(UID) --rm \
+	-v /etc/passwd:/etc/passwd:ro \
+	-v $(readlink /etc/localtime):/etc/localtime:ro \
+	--mount type=bind,source="$(PWD)",destination="/work" \
+	--mount type=volume,source=istio-go-mod,destination="/go/pkg/mod" \
+	--mount type=volume,source=istio-go-cache,destination="/gocache" \
+	--mount type=bind,source="$(GOBIN_SOURCE)",destination="/go/out/bin" \
+	-w /work $(IMG)
```

Comment:
Did you test this ? How can it work without priviledged ? The tests are creating a kind cluster, AFAIK it still needs priv.

And since the base image is different and creates other containers - not sure this is the most efficient solution

Reply:
this has been tested only for non-kind operational modes.  For this particular repository the common implementation will need work.

Reply:
> Did you test this ? How can it work without priviledged ? The tests are creating a kind cluster, AFAIK it still needs priv.
> 
> And since the base image is different and creates other containers - not sure this is the most efficient solution

Costin,
For more context we are changing out the base image currently in use in CI/CD with the image at https://gcr.io/istio-testing/build-tools which is built from this dockerfile: https://github.com/istio/tools/blob/master/docker/build-tools/Dockerfile.  We have generated a common build system that requires only docker and make in order to operate and uses pinned versions of all tools for repeatable build and testing.  In the case of the above linked dockerfile, there are some deficiencies in the dockrefile.  Kind is missing.  bazel is missing.

The install repo has gone to great lengths to optimize around kind and we don't wish to disturb that.  This functionality defaults to off, although personally, I wouldn't merge the work until this common work is functional with this repo (proxy is in a similar situation).

Cheers
-steve

Reply:
Well - the install repo IS using a base image that can not only build everything go-related in a docker container, but it can also efficiently run Kind tests ( and soon multi-cluster kind tests). 

So either istio-testing is using the same image as install, or they create a better one - but I don't think it should regress, regardless if it is on by default or not.

Reply:
the "base image" is vague :).  We have a few kicking around, and we are working on making just one to deal with that is smaller but has the necessary tools.  If I had to do over again I would have studied your work in more detail.  There is still time to take this path :)

The build dockerfile however, IMO is a writeoff.  That is the part we are reinventing from first principles.  I hear you around not breaking your workflows 👍